store/pebble/pebble.go

```diff
@@ -258,12 +260,50 @@ func (s *store) Close() error {
 	return ferr
 }
 
+// Stats gathers statistics about the values indexed by Pebble store.
+// The statistic gathering can be expensive and may be slow to calculate.
+// The numbers returned are estimated, must not be interpreted as exact and will not include
+// records that are not flushed.
+func (s *store) Stats() (*indexer.Stats, error) {
+	sst, err := s.db.SSTables(pebble.WithProperties())
+	if err != nil {
+		return nil, err
+	}
+	keygen := s.p.leaseBlake3Keyer()
+	defer keygen.Close()
+
+	start, end, err := keygen.multihashesKeyRange()
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		_ = start.Close()
+		_ = end.Close()
+	}()
+	var stats indexer.Stats
+	for _, is := range sst {
+		for _, info := range is {
+			// SST file entries may overlap beyont start or end key. Include all such overlapping
+			// SST files in the count; this effectively implies an over estimation of unique
+			// multihashes. This is fine since the number of multihashes is expected to be
+			// overwhelmingly larger than provider records.
+			if s.comparer.Compare(start.buf, info.Smallest.UserKey) <= 0 ||
+				s.comparer.Compare(end.buf, info.Largest.UserKey) <= 0 {
+				stats.MultihashCount += info.Properties.NumEntries
```

Comment:
do we have a sense of how expensive this is?

Reply:
It took 13 seconds to return on dido's snapshot from 2nd of Nov. 

Expense here just means it might take time. It scans each of the SST files.